src/Uno.UI.RuntimeTests/Tests/Windows_Foundation/Collections/Given_ValueSet.cs

```diff
@@ -0,0 +1,109 @@
+﻿#nullable enable
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using Windows.Foundation.Collections;
+using System.Runtime.InteropServices;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_Foundation.Collections
+{
+	[TestClass]
+	internal class Given_ValueSet
+	{
+
+		ValueSet _testSet = new();
+		static bool _wasEventFired = false;
+
+		[TestInitialize]
+		public void Init()
+		{
+			_testSet.MapChanged += _testSet_MapChanged;
+		}
+
+		private void _testSet_MapChanged([In] IObservableMap<string, object?> sender, [In] IMapChangedEventArgs<string> @event)
+		{
+			_wasEventFired = true;
+		}
+
+		[TestCleanup]
+		public void Cleanup()
+		{
+			_testSet.MapChanged -= _testSet_MapChanged;
+			_testSet.Clear();
+		}
+
+		[TestMethod]
+		public async void When_ModificationEvent()
+		{
+
+			// .Add
+			_wasEventFired = false;
+			_testSet.Add("thisisKey", "andValue");
+			await System.Threading.Tasks.Task.Delay(100);
```

Comment:
Why did you need to add an arbitrary delay here?

Reply:
Because we want to check if _wasEventFired is changed by handler, and handler needs time to fire and change _wasEventFired variable.

Reply:
As the event is executed synchronously, it should be possible to remove all the awaits in this test